cli/azd/internal/repository/app_init.go

```diff
@@ -137,11 +137,23 @@ func (i *Initializer) InitFromApp(
 		if prj.Language == appdetect.Java {
 			var hasKafkaDep bool
 			var hasSpringCloudAzureDep bool
-			for _, dep := range prj.AzureDeps {
-				if eventHubs, ok := dep.(appdetect.AzureDepEventHubs); ok && eventHubs.UseKafka {
+			for depIndex := range prj.AzureDeps {
+				dep := &prj.AzureDeps[depIndex]
+				if eventHubs, ok := (*dep).(appdetect.AzureDepEventHubs); ok && eventHubs.UseKafka {
 					hasKafkaDep = true
+					springBootVersion := eventHubs.SpringBootVersion
+
+					if springBootVersion == appdetect.UnknownSpringBootVersion {
+						var err error
+						springBootVersion, err = promptSpringBootVersion(i.console, ctx)
+						if err != nil {
+							return err
+						}
+						eventHubs.SpringBootVersion = springBootVersion
+						prj.AzureDeps[depIndex] = eventHubs
+					}
 				}
-				if _, ok := dep.(appdetect.SpringCloudAzureDep); ok {
+				if _, ok := (*dep).(appdetect.SpringCloudAzureDep); ok {
```

Comment:
It's OK for using `dep` instead of `(*dep)` here.
Refs: https://golang.google.cn/tour/methods/7